src/runlib.rs

```diff
@@ -128,6 +128,13 @@ pub fn record_artifacts(
 /// let byproducts = run_command(&["sh", "-c", "printf hello"], Some("tests")).unwrap();
 /// ```
 pub fn run_command(cmd_args: &[&str], run_dir: Option<&str>) -> Result<BTreeMap<String, String>> {
+    // Format output into Byproduct
+    let mut byproducts: BTreeMap<String, String> = BTreeMap::new();
+
+    if cmd_args.len() == 0 {
```

Comment:
If no `cmd_args` are provided (e.g. input is `&[]`), the `let executable = cmd_args[0];` line would fail. Because of the data type of `cmd_args` is `&[&str]`, providing an empty slice as the input is definitely possible (maybe to denote for no cmd_args supplied). In that case, there would be no byproducts and an empty BTreeMap would be returned.